### PR TITLE
Pass export function with more args to hosting CLI

### PR DESCRIPTION
**Pull request by martinxu9**
### All Submissions:

- [x] Have you followed the guidelines stated in [CONTRIBUTING.md](https://github.com/reflex-dev/reflex/blob/main/CONTRIBUTING.md) file?
- [x] Have you checked to ensure there aren't any other open [Pull Requests](https://github.com/reflex-dev/reflex/pulls ) for the desired changed?

### Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

### New Feature Submission:

- [x] Does your submission pass the tests? 
- [x] Have you linted your code locally prior to submission?

### Changes To Core Features:

- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your core changes, as applicable?
- [x] Have you successfully ran tests with your changes locally?

### **After** these steps, you're ready to open a pull request.
- For decoupling the backend/frontend deployment, the CLI core logic needs the control of backend/frontend only export. Pass those are args to let hosting CLI decide what to do.
- Bump the minimum required version of CLI to 0.1.1 (TODO: before merging)
Tests:
- [x] Pin a published pre-release version of hosting CLI to test deployment (DO NOT MERGE)

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -57,7 +57,7 @@ wrapt = [
 ]
 packaging = "^23.1"
 pipdeptree = "^2.13.0"
-reflex-hosting-cli = ">=0.1.0"
+reflex-hosting-cli = "==0.1.1a2"
```
Comment (martinxu9) [inline pyproject.toml:60]: will publish the CLI pypi and change this to >=0.1.1

**Comment by martinxu9** [comment]
@Alek99 now this PR is ready

**Comment by martinxu9** [comment]
> @Alek99 now this PR is ready

@Alek99  sorry it's not, messed up something, will fix now

File: pyproject.toml (hunk 1 above)
Comment (martinxu9) [reply pyproject.toml]: this is done now.

**Review by picklelo — APPROVED** [review]

**Pull request merged**